test_aodncore/util/test_misc.py

```diff
@@ -211,6 +212,34 @@ def test_validate_relative_path_attr(self):
         except Exception as e:
             raise AssertionError("unexpected exception raised. {e}".format(e=format_exception(e)))
 
+    def test_get_fields_from_filename(self):
+        good_pattern = re.compile(r"""
+                                      ^FILE_SUFFIX_
+                                       (?P<product_code>[A-Z]{3,4})_C-
+                                       (?P<creation_date>[0-9]{8}T[0-9]{6}Z)\.
+                                       (?P<extension>nc|txt|csv)$
+                                       """, re.VERBOSE)
+        good_file = 'FILE_SUFFIX_ABC_C-20180101T000000Z.txt'
+        bad_file = 'BAD_ABC_C-20180101T000000Z.txt'
+
+        # test on filename only
+        fields = get_fields_from_filename(good_file, good_pattern)
+        self.assertEqual(fields['product_code'], 'ABC')
+
+        # test on filepath
+        fields = get_fields_from_filename(os.path.join(TEST_ROOT, good_file),
```

Comment:
I wouldn't worry about joining this to a real path because that kind of implies the file actually exists. I'd just use a dummy "/not/a/real/path" kind of thing so it's obvious that only the path matters and not an actual file.